### DISCUSSION TITLE
Next.js Application - Module not found: Can't resolve 'fs'

### DISCUSSION THREAD

**Discussion by belfed** [vercel/next.js#45103] [Help]
Hello everyone,
I'm quite new to Next.js and I played with it a bit. My application needs to read a JSON file from the file system and uses the `fs` package to do so.
All the read/write logic to manage the JSON file is stored in the `file_handler.js` file that is inside the folder `helpers` in the root folder.
`file_handler.js` exports the `FileHandler` class that is used accross the app to handle read/write operations.

```javascript
// "./helpers/file_handler.js"
import fs from 'fs';

class FileHandler {
...
}

export default FileHandler;
```

I read that Node packages can only run on server side and that I should use them inside functions that run only on server-side (#12124).

But what if I want to have a separate file that handles every operation to avoid cluttering the components and writing the same logic all over the application? How should I manage it?

Maybe I'm missing something very obvious, but it has been a while since I stumbled upon this problem, and I can't find a solution.

Thanks in advance for the help.

**Comment by GabenGar** [comment]
Just create separate folders for `server` and `browsers` utils. This way you can at least minimize the chance of environment collisions and it will be easier to debug when you stumble upon them.
Also default export + classes is the worst combo for writing modular (let alone cross-environment) code. Almost all bundlers will bail out of statically analyzing it and therefore tree-shaking.
